clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/view/DcosJobProvider.groovy

```diff
@@ -20,16 +22,24 @@ class DcosJobProvider implements JobProvider<DcosJobStatus> {
 
   private final DcosClientProvider dcosClientProvider
   private final ObjectMapper objectMapper
+  private final AccountCredentialsProvider credentialsProvider
 
   @Autowired
-  DcosJobProvider(DcosClientProvider dcosClientProvider, ObjectMapper objectMapper) {
+  DcosJobProvider(DcosClientProvider dcosClientProvider, ObjectMapper objectMapper, AccountCredentialsProvider credentialsProvider) {
     this.dcosClientProvider = dcosClientProvider
     this.objectMapper = objectMapper
+    this.credentialsProvider = credentialsProvider
   }
 
   @Override
   DcosJobStatus collectJob(String account, String location, String id) {
-    def dcosClient = dcosClientProvider.getDcosClient(account, location)
+    def credentials = credentialsProvider.getCredentials(account)
+
+    if (!(credentials instanceof DcosAccountCredentials)) {
```

Comment:
* It'd be good to comment why this check is necessary because it's probably not obvious why this method could get called with a non-DC/OS account
* Are there any other places that we get credentials by account name that could have a problem?